diff --git a/common/DcgmStringHelpers.cpp b/common/DcgmStringHelpers.cpp
--- a/common/DcgmStringHelpers.cpp
+++ b/common/DcgmStringHelpers.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include "DcgmStringHelpers.h"
+#include <algorithm>

 #include <cstring>
 #include <set>
diff --git a/common/EntityListHelpers.cpp b/common/EntityListHelpers.cpp
--- a/common/EntityListHelpers.cpp
+++ b/common/EntityListHelpers.cpp
@@ -108,7 +108,7 @@ EntityMapTypedIterator() = default;

     bool operator==(EntityMapTypedIterator const &other) const
     {
-        return m_it == other.it;
+        return m_it == other.m_it;
     }

     template <typename Y>
diff --git a/nvvs/include/IgnoreErrorCodesHelper.h b/nvvs/include/IgnoreErrorCodesHelper.h
--- a/nvvs/include/IgnoreErrorCodesHelper.h
+++ b/nvvs/include/IgnoreErrorCodesHelper.h
@@ -22,6 +22,7 @@
 #include <exception>
 #include <map>
+#include <optional>
 #include <string>
 #include <unordered_set>
