diff --git a/pkgs/development/python-modules/autobahn/default.nix b/pkgs/development/python-modules/autobahn/default.nix
--- a/pkgs/development/python-modules/autobahn/default.nix
+++ b/pkgs/development/python-modules/autobahn/default.nix
@@ -9,8 +9,12 @@
 
   # dependencies
+  cbor2,
   cryptography,
   hyperlink,
+  msgpack,
+  py-ubjson,
   pynacl,
   txaio,
+  ujson,
 
   # optional-dependencies
   # compress
@@ -23,12 +27,6 @@
   # scram
   argon2-cffi,
   passlib,
-  # serialization
-  cbor2,
-  flatbuffers,
-  msgpack,
-  ujson,
-  py-ubjson,
   # twisted
   attrs,
   twisted,
@@ -64,10 +62,14 @@ buildPythonPackage (finalAttrs: {
   ];
 
   dependencies = [
+    cbor2
+    cffi
     cryptography
     hyperlink
+    msgpack
+    py-ubjson
     pynacl
     txaio
+    ujson
   ];
 
   optional-dependencies = lib.fix (self: {
@@ -98,13 +100,7 @@ buildPythonPackage (finalAttrs: {
       cffi
       passlib
     ];
-    serialization = [
-      cbor2
-      flatbuffers
-      msgpack
-      ujson
-      py-ubjson
-    ];
+    serialization = [ ];
     twisted = [
       attrs
       twisted
@@ -120,8 +116,7 @@ buildPythonPackage (finalAttrs: {
     pytestCheckHook
   ]
   ++ finalAttrs.passthru.optional-dependencies.encryption
-  ++ finalAttrs.passthru.optional-dependencies.scram
-  ++ finalAttrs.passthru.optional-dependencies.serialization;
+  ++ finalAttrs.passthru.optional-dependencies.scram;
 
   preCheck = ''
     # Run asyncio tests (requires twisted)
diff --git a/pkgs/development/python-modules/opentelemetry-exporter-otlp-proto-grpc/default.nix b/pkgs/development/python-modules/opentelemetry-exporter-otlp-proto-grpc/default.nix
--- a/pkgs/development/python-modules/opentelemetry-exporter-otlp-proto-grpc/default.nix
+++ b/pkgs/development/python-modules/opentelemetry-exporter-otlp-proto-grpc/default.nix
@@ -7,6 +7,7 @@
   opentelemetry-api,
   opentelemetry-exporter-otlp-proto-common,
   opentelemetry-proto,
+  opentelemetry-sdk,
   opentelemetry-test-utils,
   pytestCheckHook,
 }:
@@ -26,6 +27,7 @@ buildPythonPackage {
     opentelemetry-api
     opentelemetry-exporter-otlp-proto-common
     opentelemetry-proto
+    opentelemetry-sdk
   ];
 
   nativeCheckInputs = [
diff --git a/pkgs/development/python-modules/opentelemetry-instrumentation/default.nix b/pkgs/development/python-modules/opentelemetry-instrumentation/default.nix
--- a/pkgs/development/python-modules/opentelemetry-instrumentation/default.nix
+++ b/pkgs/development/python-modules/opentelemetry-instrumentation/default.nix
@@ -5,6 +5,7 @@
   fetchFromGitHub,
   hatchling,
   opentelemetry-api,
+  opentelemetry-semantic-conventions,
   opentelemetry-test-utils,
   pytestCheckHook,
   setuptools,
@@ -30,6 +31,7 @@ buildPythonPackage (finalAttrs: {
 
   dependencies = [
     opentelemetry-api
+    opentelemetry-semantic-conventions
     setuptools
     wrapt
   ];
diff --git a/pkgs/development/python-modules/opentelemetry-util-http/default.nix b/pkgs/development/python-modules/opentelemetry-util-http/default.nix
--- a/pkgs/development/python-modules/opentelemetry-util-http/default.nix
+++ b/pkgs/development/python-modules/opentelemetry-util-http/default.nix
@@ -2,6 +2,7 @@
   buildPythonPackage,
   hatchling,
   opentelemetry-instrumentation,
+  opentelemetry-semantic-conventions,
   opentelemetry-test-utils,
   pytestCheckHook,
 }:
@@ -14,6 +15,8 @@ buildPythonPackage {
 
   build-system = [ hatchling ];
 
+  dependencies = [ opentelemetry-semantic-conventions ];
+
   nativeCheckInputs = [
     opentelemetry-instrumentation
     opentelemetry-test-utils
diff --git a/pkgs/development/python-modules/sse-starlette/default.nix b/pkgs/development/python-modules/sse-starlette/default.nix
--- a/pkgs/development/python-modules/sse-starlette/default.nix
+++ b/pkgs/development/python-modules/sse-starlette/default.nix
@@ -39,6 +39,7 @@ buildPythonPackage rec {
 
   dependencies = [
     anyio
+    starlette
   ];
 
   optional-dependencies = {
